oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/web/DefaultBearerTokenResolver.java

```diff
@@ -87,7 +89,7 @@ public void setAllowUriQueryParameter(boolean allowUriQueryParameter) {
 
 	private static String resolveFromAuthorizationHeader(HttpServletRequest request) {
 		String authorization = request.getHeader(HttpHeaders.AUTHORIZATION);
-		if (StringUtils.hasText(authorization) && authorization.startsWith("Bearer")) {
+		if (StringUtils.hasText(authorization) && authorization.toLowerCase().startsWith("bearer")) {
```

Comment:
It would be more efficient if we didn't lowercase the entire string just to compare the first few characters.

Could we instead do

```suggestion
		if (StringUtils.hasText(authorization) && authorization.regionMatches(true, 0, "bearer", 0, 6)) {
```

Or

```java
if (StringUtils.hasText(authorization) && startsWithIgnoreCase(authorization, "bearer"))

...

private static boolean startsWithIgnoreCase(String header, String prefix) {
    return header.regionMatches(true, 0, prefix, 0, prefix.length);
}
```

For readability.

Reply:
Thanks! I've switched to `StringUtils.startsWithIgnoreCase` as suggested.